// @typescript-eslint/typescript-estree/dist/create-program/shared.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const path_1 = __importDefault(require("path"));
const typescript_1 = __importDefault(require("typescript"));
/**
 * Default compiler options for program generation from single root file
 */
const DEFAULT_COMPILER_OPTIONS = {
    allowNonTsExtensions: true,
    allowJs: true,
    checkJs: true,
    noEmit: true,
};
exports.DEFAULT_COMPILER_OPTIONS = DEFAULT_COMPILER_OPTIONS;
// typescript doesn't provide a ts.sys implementation for browser environments
const useCaseSensitiveFileNames = typescript_1.default.sys !== undefined ? typescript_1.default.sys.useCaseSensitiveFileNames : true;
const getCanonicalFileName = useCaseSensitiveFileNames
    ? (filePath) => path_1.default.normalize(filePath)
    : (filePath) => path_1.default.normalize(filePath).toLowerCase();
exports.getCanonicalFileName = getCanonicalFileName;
function getTsconfigPath(tsconfigPath, extra) {
    return getCanonicalFileName(path_1.default.isAbsolute(tsconfigPath)
        ? tsconfigPath
        : path_1.default.join(extra.tsconfigRootDir || process.cwd(), tsconfigPath));
}
exports.getTsconfigPath = getTsconfigPath;
function canonicalDirname(p) {
    return path_1.default.dirname(p);
}
exports.canonicalDirname = canonicalDirname;
